client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -4,27 +4,70 @@
 import React, { FunctionComponent, useState } from 'react';
 import { __ as NO__ } from '@wordpress/i18n';
 import { Button, Popover } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
 
 /**
  * Internal dependencies
  */
 import DomainPicker from './list';
+import { STORE_KEY as DOMAIN_STORE } from '../../stores/domain-suggestions';
+import { STORE_KEY as ONBOARD_STORE } from '../../stores/onboard';
+import { isFilledFormValue } from '../../stores/onboard/types';
+
+import './style.scss';
 
 const DomainPickerButton: FunctionComponent = () => {
+	// // User can search for a domain
+	const [ domainSearch, setDomainSearch ] = useState( '' );
+
 	const [ isDomainPopoverVisible, setDomainPopoverVisibility ] = useState(
 		true /* @TODO: should be `false` by default, true for dev */
 	);
 
+	// Without user search, we can provide recommendations based on title + vertical
+	const { siteTitle, siteVertical } = useSelect( select => select( ONBOARD_STORE ).getState() );
+
+	let search = domainSearch.trim();
+	if ( ! search && isFilledFormValue( siteTitle ) ) {
+		search = siteTitle;
+	}
+
+	const suggestions = useSelect(
+		select => {
+			if ( search ) {
+				return select( DOMAIN_STORE ).getDomainSuggestions( search, {
+					include_wordpressdotcom: true,
+					...( isFilledFormValue( siteVertical ) && { vertical: siteVertical.id } ),
+				} );
+			}
+		},
+		[ search, siteVertical ]
+	);
+
+	// Free .wordpress.com subdomain at index 0,
+	// Best paid domain match at index 1.
+	const [ freeDomainSuggestion, ...paidDomainSuggestions ] = suggestions ?? [];
```

Comment:
I'm skeptical of trusting the endpoint to always return in this format. How about partitioning on the `is_free` property? (this is missing from the type right now, I'll add it in another PR)

Reply:
Okay, we can do that :+1:

Reply:
(In a follow-up PR, as per Slack convo)